Add chrome_android data

Elchi3 opened the pull request:
The build will fail here until we've fixed the chrome_android versions in the data. I'm opening it to discuss the correct chrome_android versions.

Sources are: https://en.wikipedia.org/wiki/Google_Chrome_version_history and the posts on https://chromereleases.googleblog.com

Note that there has been Chrome for Android 18, then 25, and then 26-66.

Once we merge this, the current open PRs and future PRs shouldn't commit invalid Chrome for Android versions anymore as the build will fail otherwise.

This fixes https://github.com/mdn/browser-compat-data/issues/1353 and another item on https://github.com/mdn/browser-compat-data/issues/591.

CC'ing @jpmedley

jpmedley commented:
I've made a few notes in the linked tickets. As far as the missing dates, I'm not sure I can prioritize getting information on release that old. I don't believe Google supports them either.

patrickkettner commented:
[here is the missing chrome version info](https://gist.github.com/patrickkettner/4b7735e266417c2651ffee49cb76bd34). The issue is that a most of the failures are around chrome's android version (e.g. chrome 4.4) rather than an actual chrome version. those could be aliased against the linked data, but I dont know the mappings

jpmedley commented:
The new info looks good to me.

connorshea commented:
@Elchi3 any chance this will be merged? :)

Elchi3 commented:
It will be at some point. This adds additional validation to the whole repo and there are many open pull request that would need rebasing against stronger validation. I'm working on getting the pull request backlog cleared first. Thanks for your patience.

Elchi3 commented:
> here is the missing chrome version info. The issue is that a most of the failures are around chrome's android version (e.g. chrome 4.4) rather than an actual chrome version. those could be aliased against the linked data, but I dont know the mappings

That data looks like Chrome desktop releases, note that this PR is about Chrome Android. 

I think the releases I have here are correct. There just wasn't a Chrome Android sometimes and it only started with version 18, as stated in https://chromereleases.googleblog.com/2012/06/chrome-for-android-out-of-beta.html

I will update invalid versions in our data.

Elchi3 commented:
I've added a commit that updates all invalid versions to something that I think make sense:
- versions below 18 changed to 18. 
- versions 19-24 changed to version 25.

Not sure who can review this, but it would be nice to get merged now that the backlog is smaller.

Elchi3 commented:
The important part to review here is the first commit that adds chrome android release data:
https://github.com/mdn/browser-compat-data/pull/1356/commits/7e591e34bd813a3a21916972133ae008b7f94f48

Validation then checks against the entire compat data (version_added / version_removed) and in the second commit I've corrected invalid data that was found. I don't think these changes need special reviewing, because the tests would catch mistakes there. The PR is rebased against the current master.

Elchi3 commented:
@wbamberg would you review this for me?
In return, feel free to bug me about a review elsewhere :slightly_smiling_face:

jpmedley commented:
By the way, I suggest NOT listing dates for unreleased versions of Chrome until they are actually released. We generally keep to our six-week schedule, but delays are not unheard of.

wbamberg commented on browsers/chrome_android.json:
```diff
@@ -0,0 +1,240 @@
+          "status": "retired"
+        },
+        "26": {
+          "release_date": "2013-03-03",
```
-> "2013-04-03"

wbamberg commented on browsers/chrome_android.json:
```diff
@@ -0,0 +1,240 @@
+        },
+        "48": {
+          "release_date": "2016-01-26",
+          "release_notes": "https://chromereleases.googleblog.com/2016/01/chrome-for-android-update.html",
```
There's no blog post here - it seems there wasn't a blog post for this release.

Elchi3 replied:
I can't find one either. I'm removing this.

wbamberg commented on browsers/chrome_android.json:
```diff
@@ -0,0 +1,240 @@
+          "status": "current"
+        },
+        "69": {
+          "release_date": "2018-09-04",
```
I agree with Joe, having release dates in the future seems dodgy :).

Elchi3 commented:
Thank you @wbamberg! :tada: 

I've addressed the review comments and rebased against latest master to ensure no new data is invalid according to the now added version validation here. Let's merge this :slightly_smiling_face:
